Sources/Basic/OutputByteStream.swift

```diff
@@ -405,3 +441,89 @@ public struct Format {
         return TransformedSeparatedListStreamable(items: items, transform: transform, separator: separator)
     }
 }
+
+/// Represents a stream which operates in memory.
+public final class InMemoryOutputByteStream: OutputByteStream {
+
+    /// Contents of the stream.
+    private var contents = [UInt8]()
```

Comment:
Can you add a `FIXME` here that it would be nice to be able to share this buffer with the base class buffer? It's another level of optimization I don't think is needed in the initial change.